lib/internal/crypto/diffiehellman.js

```diff
@@ -213,6 +214,30 @@ ECDH.prototype.getPublicKey = function getPublicKey(encoding, format) {
   return key;
 };
 
+ECDH.convertKey = function convertKey(key, curve, inEnc, outEnc, format) {
+  const encoding = getDefaultEncoding();
+  inEnc = inEnc || encoding;
+  outEnc = outEnc || encoding;
+  var f;
+  if (format) {
+    if (format === 'compressed')
+      f = POINT_CONVERSION_COMPRESSED;
+    else if (format === 'hybrid')
+      f = POINT_CONVERSION_HYBRID;
+    // Default
+    else if (format === 'uncompressed')
+      f = POINT_CONVERSION_UNCOMPRESSED;
+    else
+      throw new errors.TypeError('ERR_CRYPTO_ECDH_INVALID_FORMAT', format);
+  } else {
+    f = POINT_CONVERSION_UNCOMPRESSED;
+  }
+  var convertedKey = _ECDHConvertKey(toBuf(key, inEnc), curve, f);
+  if (outEnc && outEnc !== 'buffer')
+    convertedKey = convertedKey.toString(outEnc);
+  return convertedKey;
+};
```

Comment:
Don't duplicate this logic, factor it out so it can be shared with `ECDH#getPublicKey()`.  Likewise the C++ code.

Reply:
Done!